src/core/SampleBuffer.cpp

```diff
@@ -1447,3 +1457,167 @@ SampleBuffer::handleState::~handleState()
 {
 	src_delete( m_resamplingData );
 }
+
+
+
+// Fading related functions
+
+float SampleBuffer::fadingVal(f_cnt_t pos, bool is_left)
+{
+    float relativePos = static_cast<float>(pos)/static_cast<float>(m_frames);
+    return (this->*fadefuncPtr)(relativePos, is_left);
+}
+
+
+void SampleBuffer::setLeftFader(float lfader)
+{
+    m_leftFader = lfader;
+    checkFadingActive();
+}
+
+
+void SampleBuffer::setRightFader(float rfader)
+{
+    m_rightFader = rfader;
+    checkFadingActive();
+}
+
+
+void SampleBuffer::checkFadingActive()
+{
+    if ((m_leftFader > 0.0) || (m_rightFader < 1.0))
+    {
+        m_fading = true;
+    }
+    else
+    {
+        m_fading = false;
+    }
+}
+
+
+float SampleBuffer::bezierFade(float relativePos, bool is_left)
+{
+    float amp = 1.0f;
+    if (is_left)
+    {
+        amp *= m_leftBezierFade.evaluate(relativePos/m_leftFader);
+    }
+    else
+    {
+        amp *= m_rightBezierFade.evaluate(1-(relativePos-m_rightFader)/(1-m_rightFader));
+    }
+    return amp;
+}
```

Comment:
```suggestion
float SampleBuffer::bezierFade(float relativePos, bool isLeft)
{
    float amp = 1.0f;
    if (isLeft)
    {
        amp *= m_leftBezierFade.evaluate(relativePos / m_leftFader);
    }
    else
    {
        amp *= m_rightBezierFade.evaluate(1 - (relativePos - m_rightFader) / (1 - m_rightFader));
    }
    return amp;
}
```

camelCase naming and space around the arithmetic operators.